web/modules/custom/joinup_group/src/Plugin/Validation/Constraint/UniqueShortIdInsensitiveValidator.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_group\Plugin\Validation\Constraint;
+
+use Drupal\Core\Database\Database;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates that a field is unique for the given entity type within a bundle.
+ */
+class UniqueShortIdInsensitiveValidator extends ConstraintValidator {
```

Comment:
Let's add a clear warning in the class docblock that this only is intended to work for the short ID field.

Reply:
Added it to both the constraint class and the validator class.